arho_feature_template/core/plan_manager.py

```diff
@@ -893,8 +893,10 @@ def save_regulation(regulation: Regulation) -> str | None:
 def save_plan_theme_association(
     plan_theme_id: str, regulation_id: str | None = None, proposition_id: str | None = None
 ) -> bool:
-    if PlanThemeAssociationLayer.association_exists(
-        plan_theme_id=plan_theme_id, plan_regulation_id=regulation_id, plan_proposition_id=proposition_id
+    if PlanThemeAssociationLayer.regulation_association_exists(
+        plan_theme_id=plan_theme_id, plan_regulation_id=regulation_id
+    ) or PlanThemeAssociationLayer.proposition_association_exists(
+        plan_theme_id=plan_theme_id, plan_proposition_id=proposition_id
     ):
         return True
```

Comment:
Nyt jos luodaan uusi assosiaatio, ajetaan molemmat `regulation_association_exists` ja `proposition_association_exists`, eli käydään kaikki assosiaation kohteet kahdesti läpi. Suorityskyvyn kannalta olisi hyvä, jos käytäyisiin enintään kerran.

```suggestion
    if regulation_id is not None and PlanThemeAssociationLayer.regulation_association_exists(
        plan_theme_id=plan_theme_id, plan_regulation_id=regulation_id
    ):
        return True
     if  proposition_id is not None and PlanThemeAssociationLayer.proposition_association_exists(
        plan_theme_id=plan_theme_id, plan_proposition_id=proposition_id
    ):
        return True
```